lib/puppetfile-resolver/spec_searchers/git/gclone.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require 'tempfile'
+require 'English'
+require 'puppetfile-resolver/util'
+require 'puppetfile-resolver/spec_searchers/common'
+require 'puppetfile-resolver/spec_searchers/git_configuration'
+require 'uri'
+module PuppetfileResolver
+  module SpecSearchers
+    module Git
+      module GClone
+        CLONE_CMD = 'git clone --bare --depth=1 --single-branch'
+        # @summary clones the remote url and reads the metadata file
+        # @returns [String] the content of the metadata file
+        def self.metadata(puppetfile_module, resolver_ui, config)
+          repo_url = puppetfile_module.remote
+
+          return nil if repo_url.nil?
+          return nil unless valid_http_url?(repo_url)
+          metadata_file = 'metadata.json'
```

Comment:
Like you mentioned in PR comments already.  Need a guard here to check if `git` is even available.  And memoized too would be best.